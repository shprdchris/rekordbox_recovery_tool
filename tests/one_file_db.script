# Test for correct generation of an RBR XML
# for a music collection of one file
#!/bin/sh
set -e

# Set path variables
show_warning_flags=True
music_root="$DATADIR"/root
out_db=$DATADIR/one_file_db.xml
venv_path=$DATADIR/.venv

# Activate the virtual environment for python
python3 -m venv $venv_path
. $venv_path/bin/activate

# Install required dependnecies
pip install mutagen -q

# TODO Create the db, set all variables
python3 -c "import rbrDatabase; db_main =rbrDatabase.rbrDatabase();\
	db_main.add_recursively('${music_root}',${show_warning_flags});\
	db_main.write_pretty('$out_db')"

# Deactivate virtual environment
deactivate

# Cat the contents of the resulting xml

# NOTE: the XML produced on my local machine uses the 0xA newline, and
# the XML of the pipeline uses the 0xD newline.
# The safest bet is to remove the newlines altogether.
cat $out_db | sed -e "s|${DATADIR}|DATADIR|" | \
	sed -e "s|\x0D||"
