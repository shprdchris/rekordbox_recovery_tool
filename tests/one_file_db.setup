#!/bin/sh
set -e

# Create the folder structure
music_root="$DATADIR"/root
mkdir $music_root
mkdir "$music_root"/subdir_one

# Populate with files
cp tests/track_samples/chopin_1.mp3 "${music_root}"/subdir_one

# Datbase location
db_path="$DATADIR"/db_out
