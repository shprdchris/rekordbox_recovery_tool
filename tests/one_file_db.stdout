<?xml version="1.0" ?>
<DJ_PLAYLISTS Version="1.0.0">
	<PRODUCT Name="rekordbox" Version="5.8.6" Company="Pioneer DJ"/>
	<COLLECTION Entries="1">
		<TRACK TrackID="1" Name="Chopin, Etude Op. 10, No. 9" Kind="MP3 File" Size="2822400000" Artist="Leo Sirota" TrackNumber="1" Album="" Composer="Leo Sirota" Comments="URL: http://freemusicarchive.org/music/Leo_Sirota/A_Chopin_Recital/sirotachopinmp3
Comments: http://freemusicarchive.org/
Curator: Arbiter Records
Copyright: Attribution-Noncommercial-Share Alike 3.0 United States: http://creativecommons.org/licenses/by-nc-sa/3.0/us/" Year="" BitRate="64000" SampleRate="44100" TotalTime="160.19225" DiscNumber="" AverageBpm="" DateAdded="2022-03-05" PlayCount="" Rating="" Location="file://localhost/DATADIR/root/subdir_one/chopin_1.mp3"/>
	</COLLECTION>
	<PLAYLISTS>
		<NODE Name="ROOT" Type="0" Count="1" KeyType="0">
			<NODE Name="subdir_one" Type="1" Entries="1" KeyType="0">
				<TRACK Key="1"/>
			</NODE>
		</NODE>
	</PLAYLISTS>
</DJ_PLAYLISTS>
